util/src/main/java/org/killbill/billing/util/PluginProperties.java

```diff
@@ -50,13 +49,13 @@
     }
 
     public static List<PluginProperty> buildPluginProperties(@Nullable final Map<String, Object> data) {
-        final ImmutableList.Builder<PluginProperty> propertiesBuilder = ImmutableList.<PluginProperty>builder();
+        final List<PluginProperty> list = new ArrayList<>();
         if (data != null) {
             for (final String key : data.keySet()) {
                 final PluginProperty property = new PluginProperty(key, data.get(key), false);
-                propertiesBuilder.add(property);
+                list.add(property);
             }
         }
-        return propertiesBuilder.build();
+        return List.copyOf(list);
```

Comment:
For these `ImmutableList`, should we instead return a `Collections.unmodifiableList` view? That way, we keep the _Immutable_ aspect?

Reply:
Yes, I had a doubt whether List.copyOf() is immutable, but according to javadocs, it is. So I think `List.copyOf()` is enough.